### PR TITLE
Authentication support

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds support for authentication to our hubot plugin.

Right now we don't refresh a token before it expires, so it's not fully bulletproof. Eventually, we should add support for refreshing the token before expiration to st2client.js (we should do the same with the Python client, right now, tokens are refreshed by the CLI). On top of that, we should also use st2client to perform all the request which means we should be able to consolidate everything in a couple of lines of code.

### CODE REVIEW

File: scripts/st2bot.js [hunk 1]
```diff
@@ -16,22 +16,38 @@
...
+
+// Optional, if not provided, we infer it from the API URL
+env.ST2_AUTH_URL = env.ST2_AUTH_URL || null;
+
 // Command reload interval in seconds
 env.ST2_COMMANDS_RELOAD_INTERVAL = env.ST2_COMMANDS_RELOAD_INTERVAL || 120;
 env.ST2_COMMANDS_RELOAD_INTERVAL = parseInt(env.ST2_COMMANDS_RELOAD_INTERVAL, 10);
 
+function isNotNull(value) {
+  return (!value) || value === 'null';
```
Comment (manasdk) [inline scripts/st2bot.js]: this sucks. !value does not catch 'null'?

**Comment by manasdk** [comment]
Any chance we can break out the auth piece in a separate module? The way I see it all the bot needs isa token.

File: scripts/st2bot.js (hunk 1 above)
Comment (Kami) [inline scripts/st2bot.js]: Yeah, it doesn't catch a string "null".

The problem is that if the environment variable is not specified, it somehow and somewhere gets set to string null. That's not a default node.js behavior (it should be `null`, not string "null"), but I didn't spend too much digging in where that happens.

**Comment by Kami** [comment]
@manasdk Not exactly sure what you mean with that, but as the PR description says, I do think this functionality belongs into the client (in this case, the Javascript one).

**Comment by manasdk** [comment]
The entire start method function could be moved to its own module. I suspect this will make testing simpler if nothing else - still trying to work out how to test the robot adapter itself.

**Comment by manasdk** [comment]
I am ok with this change. I might end up refactoring this subsequently if testing becomes a challenge. I've notice that if we include a lot of functionality within the st2bot module it gets rather tough to test correctly.

**Comment by Kami** [comment]
Yeah, I agree. I will merge it for now and update it when refresh functionality gets moved to the st2client (https://github.com/StackStorm/st2client.js/issues/25).

**Pull request merged**